modules/battery_manager/php/testprovisioner.class.inc

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * This file implements a data provisioner to get all possible tests
+ * for the battery_manager menu page.
+ *
+ * PHP Version 7
+ *
+ * @category   Behavioural
+ * @package    Main
+ * @subpackage Imaging
+ * @author     Henri Rabalais <henri.rabalais@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+
+namespace LORIS\battery_manager;
+
+/**
+ * This class implements a data provisioner to get all possible tests
+ * for the battery_manager menu page.
+ *
+ * PHP Version 7
+ *
+ * @category   Behavioural
+ * @package    Main
+ * @subpackage Imaging
+ * @author     Henri Rabalais <henri.rabalais@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+class TestProvisioner extends \LORIS\Data\Provisioners\DBRowProvisioner
+{
+    /**
+     * Select all tests from the test_battery table that match the attribute
+     * values passed as conditions and will return an array of test objects
+     * indexed by id.
+     *
+     * @return array $tests List of Test Objects that match the query
+     */
+    public function __construct()
+    {
+        parent::__construct(
+            "SELECT b.id,
+               t.Test_name as testName,
+               b.AgeMinDays as ageMinDays,
+               b.AgeMaxDays as ageMaxDays,
+               b.Stage as stage,
+               s.SubprojectID as subproject,
+               b.Visit_label as visitLabel,
+               p.CenterID as centerId,
+               b.firstVisit,
+               b.instr_order as instrumentOrder,
+               b.Active as active
+        FROM test_battery b
+          LEFT JOIN test_names t USING (Test_name)
+          LEFT JOIN subproject s USING (SubprojectID)
+          LEFT JOIN psc p USING (CenterID)
+        ORDER BY t.Full_name",
+            array()
+        );
+    }
+
+    /**
+     * Returns an instance of a Test object for a given table row.
+     *
+     * @param array $row The database row from the LORIS Database class.
+     *
+     * @return \LORIS\Data\DataInstance An instance representing a test.
+     */
+    public function getInstance($row) : \LORIS\Data\DataInstance
+    {
+        $cid = $row['centerId'];
+        return new TestInstance($row, $cid);
```

Comment:
Since the 'centerid' is in the row, could it be removed from the parameters ? The getCenterID function of Testinstance could return $row['centerId']